sql/plan/foreign_key_editor.go

```diff
@@ -81,6 +81,17 @@ func (fkEditor *ForeignKeyEditor) IsInitialized(editors map[*ForeignKeyEditor]st
 // Update handles both the standard UPDATE statement and propagated referential actions from a parent table's ON UPDATE.
 func (fkEditor *ForeignKeyEditor) Update(ctx *sql.Context, old sql.Row, new sql.Row, depth int) error {
 	for _, reference := range fkEditor.References {
+		// Only check the reference for the columns that are updated
+		hasChange := false
+		for _, idx := range reference.RowMapper.IndexPositions {
+			if old[idx] != new[idx] {
```

Comment:
We have to use `type.Compare` here. Some values will fail even though they're the same value, like with `[]byte`, since this will compare the slice pointers instead of the contents. Especially relevant for Doltgres.